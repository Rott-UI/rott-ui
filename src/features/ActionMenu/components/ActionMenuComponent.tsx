// React Imports
import React, {FC} from 'react'

// Component Imports
import {Item, Pressable, Label, Button, List, ActionMenu} from '../..'
import {ActionMenuStyles} from '../styles'
import {ActionMenuProps, ActionModel} from '../models'

// Constant Imports

// Util and Lib Imports

/**
 * Action Menu
 *
 * @param title Başlık alanı
 * @param subTitle Alt başlık alanı
 * @param data ActionModel[] formatında veri listesi
 * @param visible ActionMenu komponentin bulunduğu yerdeki Show state i
 * @returns Action Menu Renderlanır
 *
 */
export const ActionMenuComponent: FC<ActionMenuProps> = ({
  title,
  subTitle,
  data,
  itemHeight,
  maxItem,
  separatorTotalHeight,
}) => {
  const itemCount = (data?.length ?? 0) > maxItem! ? maxItem! : data?.length ?? 0

  // Listede renderlanacak Item
  const renderItem = ({
    title: renderItemTitle,
    name: renderItemName,
    action: onAction,
  }: ActionModel) => (
    <Pressable
      testID={`action-menu-data-${renderItemName}-test-id`}
      height={itemHeight}
      justifyContentCenter
      alignItemsCenter
      onPress={() => onAction()}>
      <Label fontSize='lg' variant='grey-900' fontWeight={500}>
        {renderItemTitle}
      </Label>
    </Pressable>
  )

  return (
    <Item size='full' style={ActionMenuStyles().actionMenuContainer} alignItemsCenter>
      <Item
        width={342}
        backgroundColor={COLOURS.GREY100}
        borderTopStartRadius={!(title || subTitle) ? 12 : 0}
        borderTopEndRadius={!(title || subTitle) ? 12 : 0}
        borderBottomStartRadius={12}
        borderBottomEndRadius={12}
        marginBottom={16}>
        <List
          testID='action-menu-test-id'
          style={{minHeight: 2}}
          width={342}
          height={itemHeight! * itemCount + separatorTotalHeight!}
          data={data}
          renderItem={({item}) => renderItem(item)}
          horizontal={false}
          separatorVariant='neutral-alpha-200'
          renderSeparator
          headerSeparator
          footerSeparator
          estimatedItemSize={display.normalize(itemHeight!, 'height')}
          showsVerticalScrollIndicator={false}
          scrollEnabled={(data?.length ?? 0) > maxItem!}
        />
      </Item>

      <Button
        variant='primary'
        onPress={() => ActionMenu.hideActionMenu()}
        fontSize='xl'
        size='full'
        testID='action-menu-cancel-test-id'
        marginBottom={21}>
        {translator('COMMON.CANCEL')}
      </Button>
    </Item>
  )
}
