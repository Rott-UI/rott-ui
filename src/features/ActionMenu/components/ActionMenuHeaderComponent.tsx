// React Imports
import React, {FC} from 'react'

// Util and Lib Imports
import {Item, Label} from '../..'
import {themeConfig} from '../../../providers'

interface ActionMenuHeaderProps {
  title?: Nullable<string>
  subTitle?: Nullable<string>
}

export const ActionMenuHeaderComponent: FC<ActionMenuHeaderProps> = ({title, subTitle}) => {
  return (
    <Item size='full' alignItemsCenter>
      <Item testID='action-menu-header-test-id' width={342}>
        {(title || subTitle) && (
          <Item
            justifyContentCenter
            alignItemsCenter
            size='full'
            backgroundColor={themeConfig.colors['grey-100']}
            borderTopStartRadius={12}
            borderTopEndRadius={12}
            height={54}>
            {title && (
              <Label
                fontSize='xl'
                variant='grey-900'
                fontWeight={700}
                marginBottom={subTitle ? 4 : undefined}>
                {title}
              </Label>
            )}
            {subTitle && (
              <Label fontSize='sm' variant='grey-900' fontWeight={500}>
                {subTitle}
              </Label>
            )}
          </Item>
        )}
      </Item>
    </Item>
  )
}
