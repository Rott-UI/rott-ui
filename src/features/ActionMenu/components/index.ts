export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
