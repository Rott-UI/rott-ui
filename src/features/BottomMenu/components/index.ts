export * from './BottomMenu'
export * from './BottomMenuItem'
