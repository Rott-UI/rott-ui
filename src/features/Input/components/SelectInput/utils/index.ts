export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
