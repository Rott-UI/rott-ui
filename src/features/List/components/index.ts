export * from './List'
export * from './FavoriteSwipeComponent'
