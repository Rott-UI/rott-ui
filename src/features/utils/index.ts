export * from './commonUiTestExtension'
export * from './commonUiStyleProperties'
