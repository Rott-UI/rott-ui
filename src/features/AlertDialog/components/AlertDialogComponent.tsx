// React Imports
import React, {FC} from 'react'

// Component Imports
import {AlertDialog, Item, Label, ModalProps, Pressable, Separator} from '../..'
import {AlertDialogButtonProps} from '../models'
import {AlertDialogStyles} from '../styles'

// Constant Imports
import {COLOURS} from '../../../constants'

// Util and Lib Imports
import {display, formatMessage} from '../../../utils'
import {ActivityIndicator} from 'react-native'

interface AlertDialogProps
  extends Omit<
    ModalProps,
    | 'header'
    | 'headerTitle'
    | 'fullScreen'
    | 'headerLogo'
    | 'closeButton'
    | 'height'
    | 'backgroundColor'
    | 'panResponderBackgroundColor'
    | 'headerBackgroundColor'
    | 'slideToClose'
  > {
  id?: number
  title: string
  text?: string
  buttons?: {
    cancelButton?: AlertDialogButtonProps
    confirmButton?: AlertDialogButtonProps
  }
  showActivityIndicator?: boolean
}

export const AlertDialogComponent: FC<AlertDialogProps> = ({
  id,
  title,
  text,
  buttons = {
    cancelButton: {
      text: 'COMMON.CANCEL',
      variant: 'danger',
      onPress: undefined,
    },
    confirmButton: undefined,
  },
  showActivityIndicator,
  ...props
}) => {
  return (
    <Item
      testID='alert-dialog-test-id'
      alignItemsCenter
      justifyContentCenter
      width={display.setWidth(80)}
      borderRadius={16}
      backgroundColor={COLOURS.GREY100}
      style={AlertDialogStyles(props).InformationModalView}>
      {showActivityIndicator && (
        <Item marginTop={24}>
          <ActivityIndicator size='large' color={COLOURS.PRIMARY} />
        </Item>
      )}

      <Item justifyContentCenter alignItemsCenter paddingHorizontal={16} paddingTop={24}>
        <Label variant='grey-900' textCenter fontSize='xl' fontWeight={600}>
          {title}
        </Label>

        <Label variant='grey-900' marginTop='5%' textCenter fontSize='md' fontWeight='normal'>
          {text}
        </Label>
      </Item>

      <Separator
        opacity={0.2}
        marginTop={24}
        orientation='horizontal'
        variant='neutral-alpha-300'
        size='full'
      />

      <Item
        row
        size='full'
        backgroundColor={COLOURS.GREY100}
        height={50}
        borderBottomStartRadius={16}
        borderBottomEndRadius={16}>
        {buttons?.cancelButton && (
          <Pressable
            testID='alert-dialog-cancel-button-test-id'
            height={50}
            justifyContentCenter
            alignItemsCenter
            onPress={() => {
              !!buttons?.cancelButton?.onPress && buttons.cancelButton.onPress()
              AlertDialog.hide(id)
            }}>
            <Item>
              <Label
                variant={buttons?.cancelButton?.variant || 'danger'}
                fontWeight={500}
                textCenter>
                {formatMessage(buttons.cancelButton.text)}
              </Label>
            </Item>
          </Pressable>
        )}

        {buttons?.confirmButton && (
          <Separator opacity={0.2} orientation='vertical' size='full' variant='neutral-alpha-300' />
        )}

        {buttons?.confirmButton && (
          <Pressable
            testID='alert-dialog-confirm-button-test-id'
            height={50}
            justifyContentCenter
            alignItemsCenter
            onPress={() => !!buttons?.confirmButton?.onPress && buttons?.confirmButton?.onPress()}>
            <Label
              variant={buttons.confirmButton.variant || 'grey-100'}
              fontWeight={500}
              textCenter>
              {formatMessage(buttons.confirmButton.text)}
            </Label>
          </Pressable>
        )}
      </Item>
    </Item>
  )
}
