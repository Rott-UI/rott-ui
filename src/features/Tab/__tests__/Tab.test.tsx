// Component Imports
import {Label, Tab} from '../../../components'

// Util and Lib Imports
import {act, fireEvent, render, waitFor} from '../../../utils'
import {commonUiTestExtension} from '../../utils'

const testId = {
  tabTestId: (name: string) => `${name}-test-id`,
}

describe('Tab -> Custom Component', () => {
  it('Tab snapshotı ile eşleşmeli', () => {
    const {tabTestId} = testId
    const rendered = render(
      <Tab testID={tabTestId('tab')} isSelected={false} onLayout={jest.fn()}>
        <Label>Test</Label>
      </Tab>
    )

    expect(rendered).toMatchSnapshot()
  })

  it('Tab onPress calismali', async () => {
    const {tabTestId} = testId
    const onPressMock = jest.fn()

    const {getByTestId} = render(
      <Tab testID={tabTestId('tab')} onPress={onPressMock}>
        <Label>Test-1</Label>
      </Tab>
    )

    await act(() => {
      let tabElement = getByTestId(tabTestId('tab'))
      fireEvent.press(tabElement)
    })

    expect(onPressMock).toHaveBeenCalled()
  })

  it('Tab accessibility state degerlerinden selected olup olmadigi kontrol edilebilmeli', async () => {
    const {tabTestId} = testId
    const returnsTrueMock = jest.fn(() => true)

    const {getByTestId} = render(
      <Tab testID={tabTestId('tab')} isSelected={returnsTrueMock()}>
        <Label>Test-1</Label>
      </Tab>
    )

    let tabElement = getByTestId(tabTestId('tab'))

    expect(tabElement).toHaveAccessibilityState({selected: true})
  })

  it('Tab selected degistigi zaman AccessibilityState degerlerinden selected degeri degismeli', async () => {
    const {tabTestId} = testId
    let isSelectedMock = true

    const {getByTestId, rerender} = render(
      <Tab
        testID={tabTestId('tab')}
        isSelected={isSelectedMock}
        onPress={() => {
          isSelectedMock = !isSelectedMock
        }}>
        <Label>Test-1</Label>
      </Tab>
    )

    await act(() => {
      const tabElement = getByTestId(tabTestId('tab'))

      expect(tabElement).toHaveAccessibilityState({selected: true})

      fireEvent.press(tabElement)
    })
    rerender(
      <Tab
        testID={tabTestId('tab')}
        isSelected={isSelectedMock}
        onPress={() => {
          isSelectedMock = !isSelectedMock
        }}>
        <Label>Test-1</Label>
      </Tab>
    )

    const tabElement = getByTestId(tabTestId('tab'))

    expect(tabElement).toHaveAccessibilityState({selected: false})
  })
})

//TODO: commonUITest eklenecek
commonUiTestExtension(<Tab testID={testId.tabTestId('tab')} />, testId.tabTestId('tab'))
