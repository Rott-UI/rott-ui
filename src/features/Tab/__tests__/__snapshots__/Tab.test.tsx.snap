// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab -> Custom Component Tab snapshotı ile eşleşmeli 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": false,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onLayout={[MockFunction]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": undefined,
        "bottom": undefined,
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": "center",
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "opacity": 1,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": 307,
        "zIndex": undefined,
      }
    }
    testID="tab-test-id"
  >
    <Text
      style={
        {
          "color": "rgba(234, 239, 240, 1)",
          "fontFamily": "Markpro-Bold",
          "fontSize": 14,
        }
      }
    />
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "flexDirection": "column-reverse",
          "justifyContent": "flex-start",
          "top": 30,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "bottom": 40,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
]
`;
